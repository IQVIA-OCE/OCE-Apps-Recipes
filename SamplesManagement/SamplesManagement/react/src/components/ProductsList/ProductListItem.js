import React  from 'react';
import { Text, TouchableOpacity, View, StyleSheet } from 'react-native';
import Status from '../Status/Status';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { secondaryGreen, themeGrey, black  } from 'apollo-react-native';

const ProductListItem = ({ item, onPress }) => {
  const handleOnPress = () => {
    if (item.locked || item.selected) return;

    if (typeof onPress === 'function') {
      onPress(item);
    }
  };

  return (
    <TouchableOpacity onPress={handleOnPress}>
      <View style={styles.listItem}>
        <View>
          <Text style={styles.itemTitle}>{item.label}</Text>
          <Status status="Info" size="large" text={item.detailLabel} />
        </View>
        <View>
          {(item.locked || item.selected) && (
            <Icon
              name="check"
              size={20}
              color={
                item.locked
                  ? themeGrey[900]
                  : secondaryGreen[900]
              }
            />
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  listItem: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: themeGrey[200],
    paddingVertical: 10,
  },
  itemTitle: {
    marginBottom: 5,
    marginLeft: 5,
    color: black[300],
  },
});

export default ProductListItem;
