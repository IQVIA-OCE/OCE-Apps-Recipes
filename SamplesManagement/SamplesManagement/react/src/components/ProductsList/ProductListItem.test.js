import React from 'react';
import renderer from 'react-test-renderer';
import ProductListItem from './ProductListItem';
import { TouchableOpacity } from 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import { Icon } from 'apollo-react-native';

let item = {
    productName: 'productName',
    name: 'name',
  },
  onPress;

describe('ProductListItem', () => {
  beforeEach(() => {
    onPress = jest.fn();
  });

  it('Should render ProductListItem component', () => {
    const { getByTestId } = render(<ProductListItem item={item} onPress={onPress} />);

    fireEvent.press(getByTestId('ProductListItem'))

    expect(onPress).toBeCalled()
  });

  it('Should render ProductListItem component lock', () => {
    const { getByTestId } = render(
      <ProductListItem item={{ ...item, locked: true }} onPress={onPress} />
    );

    fireEvent.press(getByTestId('ProductListItem'))

    expect(onPress).toBeCalledTimes(0);
  });

  it('Should render ProductListItem component selected', () => {
    const { getByTestId } = render(
      <ProductListItem item={{ ...item, selected: true }} onPress={() => {}} />
    );

    expect(getByTestId('ProductListItem_Icon')).toBeTruthy();
  });
});
