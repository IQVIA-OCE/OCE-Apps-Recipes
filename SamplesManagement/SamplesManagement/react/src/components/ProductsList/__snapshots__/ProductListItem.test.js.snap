// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListItem Should render ProductListItem component 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#DADADA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 10,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#4D4D4D",
            "marginBottom": 5,
            "marginLeft": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "borderRadius": 10,
              "flexDirection": "row",
              "paddingHorizontal": 6,
              "paddingVertical": 2,
            },
            Object {
              "borderRadius": 15,
              "fontSize": 16,
              "paddingHorizontal": 8,
              "paddingVertical": 6,
            },
            Object {
              "backgroundColor": "#ecebea",
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Info
        </Text>
      </View>
    </View>
    <View />
  </View>
</View>
`;

exports[`ProductListItem Should render ProductListItem component lock 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#DADADA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 10,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#4D4D4D",
            "marginBottom": 5,
            "marginLeft": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "borderRadius": 10,
              "flexDirection": "row",
              "paddingHorizontal": 6,
              "paddingVertical": 2,
            },
            Object {
              "borderRadius": 15,
              "fontSize": 16,
              "paddingHorizontal": 8,
              "paddingVertical": 6,
            },
            Object {
              "backgroundColor": "#ecebea",
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Info
        </Text>
      </View>
    </View>
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#8D8D8D",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`ProductListItem Should render ProductListItem component selected 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#DADADA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 10,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#4D4D4D",
            "marginBottom": 5,
            "marginLeft": 5,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "borderRadius": 10,
              "flexDirection": "row",
              "paddingHorizontal": 6,
              "paddingVertical": 2,
            },
            Object {
              "borderRadius": 15,
              "fontSize": 16,
              "paddingHorizontal": 8,
              "paddingVertical": 6,
            },
            Object {
              "backgroundColor": "#ecebea",
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Info
        </Text>
      </View>
    </View>
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#009F0B",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
